sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/variantExpressions.scala

```diff
@@ -287,31 +282,64 @@ case class VariantGet(
     timeZoneId,
     zoneId)
 
-  protected override def nullSafeEval(input: Any, path: Any): Any = {
-    VariantGet.variantGet(input.asInstanceOf[VariantVal], parsedPath, dataType, castArgs)
+  protected override def nullSafeEval(input: Any, path: Any): Any = parsedPath match {
+    case Some(pp) =>
+      VariantGet.variantGet(input.asInstanceOf[VariantVal], pp, dataType, castArgs)
+    case _ =>
+      val pathValue = path.toString
+      val parsedRowPath = VariantGet.getParsedPath(pathValue, prettyName)
+      VariantGet.variantGet(input.asInstanceOf[VariantVal], parsedRowPath, dataType, castArgs)
```

Comment:
to simply the codegen implementation, I think we should push more work to the external functions instead of in the `eval` method.

How about we add an overload of `VariantGet.variantGet` that accepts the path parameter as a `UTF8String`? The new overload will take care of the path parsing and codegen just needs to call this overload.